### PR TITLE
Improve `pathname2url()` and `url2pathname()` docs

### PR DESCRIPTION
**Pull request by barneygale**
These functions have long sown confusion among Python developers. The existing documentation says they deal with URL path components, but that doesn't fit the evidence on Windows:

```python
>>> pathname2url(r'C:\foo')
'///C:/foo'
>>> pathname2url(r'\\server\share')
'////server/share'  # or '//server/share' as of quite recently
```

If these were URL path components, they would imply complete URLs like `file://///C:/foo` and `file://////server/share`. Clearly this isn't right. Yet the implementation in `nturl2path` is deliberate, and the `url2pathname()` function correctly inverts it.

On non-Windows platforms, the behaviour until quite recently is to simply quote/unquote the path without adding or removing any leading slashes. This behaviour is compatible with *both* interpretations -- 1) the value is a URL path component (existing docs), and 2) the value is everything following `file:` (this PR)

The conclusion I draw is that these functions operate on everything after the `file:` prefix, which may include an authority section. This is the only explanation that fits both the  Windows and non-Windows behaviour. It's also a better match for the function names.

----
📚 Documentation preview 📚: https://cpython-previews--127125.org.readthedocs.build/

### CODE REVIEW

**Comment by barneygale** [comment]
I think the confusion came about because the original `urllib` author understood that `file:/etc/hosts` was the correct way to express `/etc/hosts` as a file URI. They weren't wrong per se, and it's still an acceptable form. But later on, it became much more prevalent to express that path as `file:///etc/hosts`, and some folks began to use `pathname2url()` and `url2pathname()` in a slightly different way. But the new definition never gelled with the Windows implementation.

File: Doc/library/urllib.request.rst [hunk 1]
```diff
@@ -148,9 +148,14 @@ The :mod:`urllib.request` module defines the following functions:
...
 
-   Convert the pathname *path* from the local syntax for a path to the form used in
-   the path component of a URL.  This does not produce a complete URL.  The return
-   value will already be quoted using the :func:`~urllib.parse.quote` function.
+   Convert the given local path to a ``file:`` URL. This function uses
+   :func:`~urllib.parse.quote` function to encode the path. For historical
+   reasons, the return value omits the ``file:`` scheme prefix. For example::
+
+      >>> from urllib.request import pathname2url
+      >>> path = 'C:\\Windows'
```
Comment (serhiy-storchaka) [inline Doc/library/urllib.request.rst]: If use `C:\Program Files`, you can show quoting a space as %20.

File: Doc/library/urllib.request.rst [hunk 2]
```diff
@@ -148,9 +148,14 @@ The :mod:`urllib.request` module defines the following functions:
 
 .. function:: pathname2url(path)
 
-   Convert the pathname *path* from the local syntax for a path to the form used in
-   the path component of a URL.  This does not produce a complete URL.  The return
-   value will already be quoted using the :func:`~urllib.parse.quote` function.
+   Convert the given local path to a ``file:`` URL. This function uses
+   :func:`~urllib.parse.quote` function to encode the path. For historical
+   reasons, the return value omits the ``file:`` scheme prefix. For example::
```
Comment (serhiy-storchaka) [inline Doc/library/urllib.request.rst]: Please mention that this is on Windows. On other platforms you will get other result (will it cause issues with doctests?).

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**